src/main/java/com/rationaleemotions/internal/locators/Strategy.java

```diff
@@ -36,6 +36,62 @@ public By getStrategy(String locator) {
             return By.cssSelector(locator.substring(4));
         }
     },
+    /**
+     * Represents the class locating strategy.
+     */
+    CLASS {
+        @Override
+        public boolean typeMatches(String locator) {
+            return Strategy.isNotNullAndEmpty(locator) && locator.startsWith("class");
+        }
+
+        @Override
+        public By getStrategy(String locator) {
+            return By.className(locator.substring(6));
+        }
+    },
+    /**
+     * Represents the linkText locating strategy.
+     */
+    LINK_TEXT {
+        @Override
+        public boolean typeMatches(String locator) {
+            return Strategy.isNotNullAndEmpty(locator) && locator.startsWith("linkText");
+        }
+
+        @Override
+        public By getStrategy(String locator) {
+            return By.linkText(locator.substring(9));
```

Comment:
This hard-coding of the substring length is kind of non intuitive, as in it doesn't really explain how we are arriving at this number. I guess we should instead being doing something like `By.linkText(locator.substring("linkText.")`.

We could even go the extra mile and define just two methods which each implementation of `typeMatches()` and `getStrategy()` would call into... Something like below.

```java
public boolean matches(String locator, String text) {
     return Strategy.isNotNullAndEmpty(locator) && locator.toLowerCase().startsWith(text.toLowerCase());
}
```

and then refactor the `typeMatches(locator)` to just call into this utility method.

The same thing with building a `getStrategy()` method.

You can perhaps just enhance the interface the enum is implementing to add a getter for the strategy text and then use that invocation to calculate the locator.

That way we aren't repeating anything any where and its all self explanatory.

Let me know if that explains what I am talking about.